src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/UWPApplication.cs

```diff
@@ -375,16 +376,58 @@ internal string LogoPathFromUri(string uri, string theme)
-                    {
-                        // scale factors on win10: https://docs.microsoft.com/en-us/windows/uwp/controls-and-patterns/tiles-and-notifications-app-assets#asset-size-tables,
-                        { PackageVersion.Windows10, new List<int> { 100, 125, 150, 200, 400 } },
-                        { PackageVersion.Windows81, new List<int> { 100, 120, 140, 160, 180 } },
-                        { PackageVersion.Windows8, new List<int> { 100 } }
-                    };
+                {
+                    // scale factors on win10: https://docs.microsoft.com/en-us/windows/uwp/controls-and-patterns/tiles-and-notifications-app-assets#asset-size-tables,
+                    { PackageVersion.Windows10, new List<int> { 100, 125, 150, 200, 400 } },
+                    { PackageVersion.Windows81, new List<int> { 100, 120, 140, 160, 180 } },
+                    { PackageVersion.Windows8, new List<int> { 100 } }
+                };
+
+                if(DisplayName.Contains("Feedb"))
+                {
+
+                }
+
+
+                // Check for a match with target size
+                Dictionary<string, int> pathFactorPairs = new Dictionary<string, int>();
+                foreach (var factor in targetSizes)
+                {
+                    string altPathColorize = $"{prefix}.targetsize-{factor}_altform-colorize{extension}";
+                    string altPathDeviceUnPlated = $"{prefix}.targetsize-{factor}_altform-unplated_devicefamily-colorfulunplated{extension}";
+                    string altPathUnPlated = $"{prefix}.targetsize-{factor}_altform-unplated{extension}";
+                    string suffixThemePath = $"{prefix}.targetsize-{factor}_{theme}{extension}";
+                    string prefixThemePath = $"{prefix}.{theme}_targetsize-{factor}{extension}";
+                    string simplePath = $"{prefix}.targetsize-{factor}{extension}";
+
+                    paths.Add(altPathColorize);
+                    paths.Add(altPathDeviceUnPlated);
+                    paths.Add(altPathUnPlated);
+                    paths.Add(suffixThemePath);
+                    paths.Add(prefixThemePath);
+                    paths.Add(simplePath);
+
+                    pathFactorPairs.Add(altPathColorize, factor);
+                    pathFactorPairs.Add(altPathDeviceUnPlated, factor);
+                    pathFactorPairs.Add(altPathUnPlated, factor);
+                    pathFactorPairs.Add(suffixThemePath, factor);
+                    pathFactorPairs.Add(prefixThemePath, factor);
+                    pathFactorPairs.Add(simplePath, factor);
+                }
+
+                var selectedIconPath = paths.OrderBy(x => Math.Abs(pathFactorPairs.GetValueOrDefault(x) - appIconSize)).FirstOrDefault(File.Exists);
+                if (!string.IsNullOrEmpty(selectedIconPath))
+                {
+                    return selectedIconPath;
+                }
 
+                // Check for a match with scale

```

Comment:
Can you split this method into two (at least)? Here you clear your variables and do some other check - move this logic into a separate method